build failure on Fedora 41 x86_64 with gcc 14.2.1

The error is from recent changes in the past few days. The compiler suite is: gcc (GCC) 14.2.1 20240912 (Red Hat 14.2.1-3).
After downloading and unzipping the snapshot archive corresponding to  commit c406aaf:

mkdir .cmake
cd .cmake 
cmake -G Ninja -DCMAKE_C_COMPILER=/usr/bin/gcc -DCMAKE_CXX_COMPILER=/usr/bin/g++  ../ 
yields:

[openmohaa_c406aaf_build_failure_gcc_14.2.1_fedora41_x86_64_20240930.txt](https://github.com/user-attachments/files/17195887/openmohaa_c406aaf_build_failure_gcc_14.2.1_fedora41_x86_64_20240930.txt)


[208/672] Building C object code/renderer/CMakeFiles/omohrenderer.dir/tr_image.c.o
FAILED: code/renderer/CMakeFiles/omohrenderer.dir/tr_image.c.o 
/usr/bin/gcc -DUSE_INTERNAL_JPEG=1 -I/tmp/openmohaa-main/code/renderer/../sdl -I/tmp/openmohaa-main/code/renderer/../renderercommon -I/tmp/openmohaa-main/code/renderer/../jpeg-8c -I/tmp/openmohaa-main/code/qcommon/../qcommon -I/tmp/openmohaa-main/code/qcommon/../script -I/tmp/openmohaa-main/.cmake/code/qcommon/generated -I/usr/include/SDL2 -Wno-comment -Werror=return-type -fno-gnu-unique -MD -MT code/renderer/CMakeFiles/omohrenderer.dir/tr_image.c.o -MF code/renderer/CMakeFiles/omohrenderer.dir/tr_image.c.o.d -o code/renderer/CMakeFiles/omohrenderer.dir/tr_image.c.o -c /tmp/openmohaa-main/code/renderer/tr_image.c
/tmp/openmohaa-main/code/renderer/tr_image.c: In function ‘LoadDDS’:
/tmp/openmohaa-main/code/renderer/tr_image.c:1556:21: error: passing argument 1 of ‘strncmp’ from incompatible pointer type [-Wincompatible-pointer-types]
 1556 |         if (strncmp(&signature, "DDS ", 4))
      |                     ^~~~~~~~~~
      |                     |
      |                     unsigned int *
In file included from /tmp/openmohaa-main/code/renderer/../qcommon/q_shared.h:166,
                 from /tmp/openmohaa-main/code/renderer/tr_local.h:25,
                 from /tmp/openmohaa-main/code/renderer/tr_image.c:23:
/usr/include/string.h:159:33: note: expected ‘const char *’ but argument is of type ‘unsigned int *’
  159 | extern int strncmp (const char *__s1, const char *__s2, size_t __n)
      |                     ~~~~~~~~~~~~^~~~
[211/672] Building CXX object code/renderer/CMakeFiles/omohrenderer.dir/tr_ghost.cpp.o
ninja: build stopped: subcommand failed.

--- a/code/renderer/tr_image.c
+++ b/code/renderer/tr_image.c
@@ -1553,7 +1553,7 @@ static void LoadDDS(const char* name, byte** pic, int* width, int* height, qbool
 	// reject files that don't start with "DDS "
 	//
 	ri.FS_Read(&signature, sizeof(signature), handle);
-	if (strncmp(&signature, "DDS ", 4))
+	if (strncmp(signature, "DDS ", 4))
 	{
 		ri.Printf(PRINT_ALL, "File %s is not a DDS file.\n", name);
 		ri.FS_CloseFile(handle);

